src/HLSLTree.cpp

```diff
@@ -1409,139 +1409,5 @@ void HideUnusedArguments(HLSLFunction * function)
     }
 }
 
-
-class FindFunctionCallVisitor : public HLSLTreeVisitor
```

Comment:
I've checked and our shaders compile fine both with and without pruning (we frequently include redundant definitions and also have some shaders with many many entrypoints, so the resulting .metal files compile much slower, but they do compile). Additionally, I remembered the exact problem which led me to reworking this - it's in the commit description for the specific change that replaces FindFunctionCall with NeedsFunction :) FindFunctionCall was only used as a check "does the shader use tex2D/etc.", but it's possible that there's a function that uses tex2D but is not called by the shader entrypoint - without the tree pruning this function would be emitted and fail to compile since tex2D wouldn't be.